### PR TITLE
fix: populate RouteCode on operations during transaction creation

### PR DESCRIPTION
**Pull request by ClaraTersi**
## Summary

- RouteCode was wired through DB, Redis, and API response but never populated at creation time — operations always returned `routeCode: null`
- The transaction route cache from `ValidateAccountingRules` was being discarded (`_`) at both call sites, so `BuildOperations` had no way to resolve the code
- This fix captures the cache, passes it through, and resolves each operation's RouteCode from its RouteID via a new `resolveRouteCodesFromCache` function

### Core fix
- **transaction-creation-helpers.go** — added `transactionRouteCache` parameter to `BuildOperations`; added `resolveRouteCodesFromCache` that scans Source/Destination/Bidirectional maps per action to populate `RouteCode` on each operation
- **transaction-creation-helpers.go** — capture `routeCache` from `GetBalances` (was `_`) and pass it to `BuildOperations`
- **transaction-state-handlers.go** — same: capture and pass `routeCache` for the commit/cancel/revert path

### Redis consumer
- **redis.consumer.go** — fetch route cache via `GetOrCreateTransactionRouteCache` in the write-behind consumer path; added debug-level logging when UUID parse or cache retrieval fails

### Housekeeping
- Renamed `transaction_creation_helpers.go` → `transaction-creation-helpers.go` and `transaction_state_handlers.go` → `transaction-state-handlers.go` (kebab-case per project convention)
### Tests
- 7 new unit tests for `resolveRouteCodesFromCache` covering nil cache, nil/empty RouteID, source/destination/bidirectional resolution, multiple operations, and unknown route IDs

### CODE REVIEW

**Comment by lerian-studio** [comment]
## 📊 Unit Test Coverage Report: `midaz-transaction`

| Metric | Value |
|--------|-------|
| **Overall Coverage** | `85.4%` ✅ PASS |
| **Threshold** | `85%` |

## Coverage by Package

| Package | Coverage |
|---------|----------|
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/grpc/in` | 100.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/http/in` | 78.4% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/mongodb` | 66.7% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/assetrate` | 100.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/balance` | 75.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/operation` | 90.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/operationroute` | 100.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/transaction` | 99.1% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/postgres/transactionroute` | 100.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/rabbitmq` | 93.1% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/adapters/redis/balance` | 100.0% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/services/command` | 90.3% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/services/query` | 95.2% |
| `github.com/LerianStudio/midaz/v3/components/transaction/internal/services` | 100.0% |


---
*Generated by Go PR Analysis workflow*

File: components/transaction/internal/bootstrap/redis.consumer.go [hunk 1]
```diff
@@ -389,10 +390,26 @@ func (r *RedisQueueConsumer) processMessage(ctx context.Context, key string, m m
...
+			trID, parseErr := uuid.Parse(m.Validate.TransactionRoute)
+			if parseErr != nil {
+				logger.Log(ctx, libLog.LevelDebug, fmt.Sprintf("Failed to parse TransactionRoute UUID %s: %v", m.Validate.TransactionRoute, parseErr))
+			} else {
+				cache, cacheErr := r.TransactionHandler.Query.GetOrCreateTransactionRouteCache(msgCtxWithSpan, m.OrganizationID, m.LedgerID, trID)
+				if cacheErr != nil {
+					logger.Log(ctx, libLog.LevelDebug, fmt.Sprintf("Failed to get route cache for org=%s ledger=%s route=%s: %v", m.OrganizationID, m.LedgerID, trID, cacheErr))
+				} else {
+					routeCache = &cache
+				}
```
[the remark this answers is not in the document]
Comment (ClaraTersi) [reply components/transaction/internal/bootstrap/redis.consumer.go:405]: Won't fix — intentional design. This consumer is a write-behind path: balances have already been atomically updated in Redis by the time we get here.

**Review by gandalf-at-lerian — APPROVED** [review]
Clean fix. The root cause is clear — route cache was being discarded at both `GetBalances` call sites, so `RouteCode` could never be populated.

`resolveRouteCodesFromCache` is straightforward: iterates Source → Destination → Bidirectional per action, breaks on first match. Correct since routeIDs are unique across the maps.

Redis consumer path: graceful degradation on cache miss (debug log + continue with nil) is the right call — write-behind shouldn't hard-fail because route codes couldn't resolve. The operations are still valid, just without the code decoration.

Tests cover the edge cases well (nil cache, nil/empty RouteID, all three map types, mixed resolution, unknown IDs).

One minor note: kebab-case renames mixed into the same PR — ideally separate commits, but not a blocker.

LGTM ✅
